File: set-matrix-zeroes.ts
{
    function setZeroes(matrix: number[][], cols = matrix[0].length, rows = matrix.length): void {
        const col  = Array(cols).fill(false);
        const row  = Array(rows).fill(false);
    
        for (let i = 0; i < rows; i++) 
            for (let j = 0; j < cols; j++)
                if (matrix[i][j] === 0) {
                    col[i] = true;
                    row[j] = true;
                }

        for (let i = 0; i < rows; i++)
            for (let j = 0; j < cols; j++) 
                if (col[i] || row[j]) 
                    matrix[i][j] = 0

    };

    console.log(setZeroes([[1,1,1],[1,0,1],[1,1,1]]))
}

File: set-matrix-zeroes.rs
pub fn set_zeroes(matrix: &mut Vec<Vec<i32>>) {
    let rows = matrix.len();
    if rows == 0 {
        return;
    }
    let cols = matrix[0].len();

    let mut row = vec![false; rows];
    let mut col = vec![false; cols];

    for i in 0..rows {
        for j in 0..cols {
            if matrix[i][j] == 0 {
                row[i] = true;
                col[j] = true;
            }
        }
    }

    for i in 0..rows {
        for j in 0..cols {
            if row[i] || col[j] {
                matrix[i][j] = 0;
            }
        }
    }
}
